controllers/helper_test.go

```diff
@@ -538,6 +540,52 @@ func testBuildManagedZone(name, ns, domainName string) *kuadrantdnsv1alpha1.Mana
 	}
 }
 
+// testEndpointsTraversable consumes an array of endpoints and returns a boolean
+// indicating presence of that path from host to all IPs
+func testEndpointsTraversable(endpoints []*endpoint.Endpoint, host string, ips []string) bool {
+	for _, ip := range ips {
+		allTargetsFound := false
+		for _, ep := range endpoints {
+			// the target on the endpoint exists as an endpoint
+			if ep.DNSName == host {
+				// target is our IP
+				if slices.Contains(ep.Targets, ip) {
+					return true
+				} else {
+					// target is not out IP. Use target as a host and check for existence of Endpoints with such a DNSName
+					for _, target := range ep.Targets {
+						// if at least one returns as true allTargetsFound will be locked in true
+						allTargetsFound = allTargetsFound || testEndpointsTraversable(endpoints, target, []string{ip})
+					}
+				}
+			}
+		}
+		// at least one endpoint that matches host targeted IP
+		return allTargetsFound
```

Comment:
Does this not return on the first found target?

Reply:
No, it returns "at least one emdpoint has IP" after we went though the list of all endpoints

Reply:
@mikenairn are you ok with the changes made?